libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/prompts/UserTokenAccess.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MT License.
+
+package com.microsoft.bot.dialogs.prompts;
+
+import com.microsoft.bot.builder.ConnectorClientBuilder;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.connector.ConnectorClient;
+import com.microsoft.bot.connector.authentication.ClaimsIdentity;
+import com.microsoft.bot.connector.authentication.ConnectorFactory;
+import com.microsoft.bot.connector.authentication.UserTokenClient;
+import com.microsoft.bot.schema.SignInResource;
+import com.microsoft.bot.schema.TokenExchangeRequest;
+import com.microsoft.bot.schema.TokenResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class UserTokenAccess {
+
+    private UserTokenAccess() {
+    }
```

Comment:
Is this constructor really needed? Java will create the empty constructor automatically if it does not exist

Reply:
as far as I saw, Java requires a private constructor to make the class final. I'll doublecheck